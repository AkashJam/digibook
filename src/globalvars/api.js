const path = "http://10.0.2.2:3000";

const API = {
  // user
  login: async (user) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  register: async (user) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/register`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateUser: async (userdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(userdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },

  // groups
  getGroups: async (userID) => {
    //GET request
    const response = await fetch(`${path}/group/${userID}`, {
      method: "GET",
      //Request Type
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        //Success
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        //Error
        return error;
      });
    return response;
  },
  createGroup: async (groupdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/group/add`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(groupdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateGroup: async (groupdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/group/update`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(groupdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },

  // Tasks
  getTasks: async (userID) => {
    //GET request
    const response = await fetch(`${path}/task/${userID}`, {
      method: "GET",
      //Request Type
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        //Success
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        //Error
        return error;
      });
    return response;
  },
  createTask: async (taskdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/task/add`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(taskdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateTask: async (taskdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/task/update`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(taskdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
};

export default API;
