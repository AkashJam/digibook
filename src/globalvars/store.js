import React, { useReducer } from "react";
import { toastr } from "./";
import AsyncStorage from "@react-native-async-storage/async-storage";

const user = {
  id: 0,
  username: "",
  displayName: null,
  range: 0.75,
  collaborators: [],
  groups: [],
  activities: [],
  notifications: [],
  locations: {},
};

const storeUserLog = (data) =>
  AsyncStorage.setItem(`DN_userlog`, JSON.stringify(data));

const reducer = (state, action) => {
  let log = [];
  let obj = {};
  switch (action.type) {
    //users

    case "set_user":
      // console.log(action.locations);
      obj = {
        id: action.user.id,
        username: action.user.username,
        displayName: action.user.displayName,
        range: action.user.range,
        collaborators: action.user.collaborators,
        groups: action.user.groups,
        activities: action.user.activities,
        notifications: action.notifications,
        locations: action.locations,
      };
      storeUserLog(obj);
      toastr(
        `Welcome ${
          action.user.displayName !== null
            ? action.user.displayName
            : action.user.username
        }`
      );
      return obj;

    case "update_display_name":
      // obj = await API.updateUser({
      //   id: state.id,
      //   task: { display_name: action.displayName },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: action.displayName,
        range: state.range,
        collaborators: action.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: state.notifications,
        locations: state.locations,
      };
      storeUserLog(obj);
      toastr("Display name updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    case "update_user_range":
      // obj = await API.updateUser({
      //   id: state.id,
      //   user: { range: action.range },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: state.displayName,
        range: action.range,
        collaborators: state.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: state.notifications,
        locations: state.locations,
      };
      storeUserLog(obj);
      toastr("Notification range updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    //activities
    case "add_task":
      const newTask = {
        id: action.id,
        description: action.description,
        group_id: action.group_id,
        notify: false,
        location: null,
        datetime: action.datetime !== undefined ? action.datetime : null,
        completed: false,
        createdOn: new Date().toString(),
        modifiedOn: null,
      };
      log = [newTask, ...state.activities];
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task added.");
      return obj;

    case "rename_task":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            description: action.description,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task renamed.");
      return obj;

    case "delete_task":
      log = state.activities.filter((activity) => activity.id !== action.id);
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task deleted.");
      return obj;

    case "change_group":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            group_id: action.group_id,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Group changed.");
      return obj;

    case "change_datetime":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            datetime: action.datetime,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Date and time changed.");
      return obj;

    case "change_location":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            location: action.location,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Location changed.");
      return obj;

    case "toggle_notification":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            notify: action.notify,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr(action.notify ? "Notification On." : "Notification Off.");
      return obj;

    case "toggle_completion":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            completed: action.completed,
            datetime: new Date().toString(),
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr(
        action.completed ? "Task completed." : "Task no longer completed."
      );
      return obj;

    // groups
    case "add_group":
      let newGroup = {
        id: action.id,
        name: action.name,
        createdOn: new Date().toString(),
        modifiedOn: null,
      };
      log = [newGroup, ...state.groups];
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group added.");
      return obj;

    case "rename_group":
      // obj = await API.updateGroup({
      //   id: action.id,
      //   group: { name: action.name },
      // });
      // if (obj.code == 200) {
      log = state.groups.map((group) => {
        if (group.id === action.id) {
          return {
            ...group,
            name: action.name,
            modifiedOn: new Date().toString(),
          };
        } else {
          return group;
        }
      });
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group renamed.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    case "delete_group":
      // obj = await API.updateGroup({
      //   id: action.id,
      //   group: { active: false },
      // });
      // if (obj.code == 200) {
      log = state.groups.filter((group) => group.id !== action.id);
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group deleted.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    //notifications
    case "update_notifications":
      // obj = await API.updateUser({
      //   id: state.id,
      //   task: { display_name: action.displayName },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: state.displayName,
        range: state.range,
        collaborators: state.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: action.notifications,
        locations: state.locations,
      };
      // storeUserLog(obj);
      // toastr("Display name updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    // locations
    case "set_location":
      obj = state.locations ? state.locations : {};
      if (
        !state.locations ||
        state.locations.lat === null ||
        state.locations.lon === null
      ) {
        obj.lat = action.latitude;
        obj.lon = action.longitude;
      } else if (
        Math.abs(obj.latitude - action.latitude) > 0.01 &&
        Math.abs(obj.longitude - action.longitude) > 0.01
      ) {
        Object.keys(obj).forEach((e) => {
          if (e !== "lat" && e !== "lon") obj[e] = [];
        });
        obj.lat = action.latitude;
        obj.lon = action.longitude;
      }
      obj[`${action.use}`] = action.location;
      storeUserLog({ ...state, locations: obj });
      return { ...state, locations: obj };

    default:
      return state;
  }
};

const UserContext = React.createContext({
  state: user,
  dispatch: () => null,
});

const UserProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, user);

  return (
    <UserContext.Provider value={[state, dispatch]}>
      {children}
    </UserContext.Provider>
  );
};

export { UserContext, UserProvider };
