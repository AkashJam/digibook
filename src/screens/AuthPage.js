import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  TextInput,
  Dimensions,
} from "react-native";
import {
  COLORS,
  SIZES,
  FONTS,
  SHADOW,
  PAGE,
  PAGEHEAD,
  TOAST,
} from "../constants";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { toastr, UserContext } from "../globalvars";
import { ActivityIndicator } from "react-native";

export default function AuthPage({ navigation }) {
  const [state, dispatch] = React.useContext(UserContext);

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const [canRegister, setCanRegister] = useState(false);
  const [canLogin, setCanLogin] = useState(false);

  const [userColor, setUserColor] = useState(COLORS.primary);
  const [pwdColor, setPwdColor] = useState(COLORS.primary);

  const userValidation = () => {
    let validUser,
      validPwd = false;

    //Border validation
    if (username !== "") {
      if (username.length < 3 || username.length > 20) setUserColor(COLORS.red);
      else {
        setUserColor(COLORS.green);
        validUser = true;
      }
    } else setUserColor(COLORS.primary);

    if (password !== "") {
      if (password.length < 8 || password.length > 20) setPwdColor(COLORS.red);
      else {
        setPwdColor(COLORS.green);
        validPwd = true;
      }
    } else setPwdColor(COLORS.primary);

    //Button validation
    if (validUser && validPwd) {
      setCanLogin(true);
      setCanRegister(true);
    } else {
      setCanLogin(false);
      setCanRegister(false);
    }
  };

  const [isLoading, setLoading] = useState(false);

  const login = async () => {
    const check = setTimeout(() => {
      // Add notification that user does not exist and you can create a user with that name
      setLoading(false);
      toastr("User does not exist.", 5000);
      setCanLogin(false);
    }, 5000); //Timeout if no reseponse for 5 sec
    setLoading(true);
    //Need to check with api if user exists and login else
    const user = await AsyncStorage.getItem(username); //Check local storage for user

    if (user !== null) {
      clearTimeout(check);
      setLoading(false);
      //if wrong password
      if (password !== user) {
        setCanLogin(false);
        setCanRegister(false);
        setUserColor(COLORS.red);
        setPwdColor(COLORS.red);
        toastr("Wrong username or password.", 5000);
      } else {
        let log = await AsyncStorage.getItem(`${username}_log`); //Check local storage for user;
        let obj = {
          username: username,
          activities: log != null ? JSON.parse(log) : [],
        };
        dispatch({ type: "set_user", user: obj });
        toastr(`Welcome ${username}`, 5000);
        setUsername("");
        setPassword("");
        navigation.navigate("Home");
      }
    }
  };

  const register = async () => {
    const check = setTimeout(async () => {
      setLoading(false);
      try {
        await AsyncStorage.setItem(username, password);
        await AsyncStorage.setItem(`${username}_log`, JSON.stringify([]));
        setCanRegister(false);
        setCanLogin(true);
        toastr("You have been registered.", 5000);
      } catch (e) {
        toastr("Registration error.", 5000);
      }
    }, 5000);
    setLoading(true);
    //Need to check with api if user exists and login else
    const user = await AsyncStorage.getItem(username); //Check local storage for user
    //if user exists
    if (user !== null) {
      clearTimeout(check);
      //Notification that username exists
      setLoading(false);
      toastr("Username already exists.", 5000);
      setCanRegister(false);
    }
  };

  useEffect(() => {
    userValidation(); //Set highlight of border and buttons
  }, [username, password]);

  return (
    <View style={styles.page}>
      {isLoading && (
        <View style={styles.activity}>
          <ActivityIndicator size={100} color={COLORS.primary} />
        </View>
      )}
      <Text style={PAGEHEAD}>Digital Notebook</Text>
      <TextInput
        style={{ ...styles.textInput, borderColor: userColor }}
        placeholder="Username"
        placeholderTextColor={COLORS.secondary}
        onChangeText={(text) => setUsername(text)}
        value={username}
      />
      <TextInput
        style={{ ...styles.textInput, borderColor: pwdColor }}
        placeholder="Password"
        placeholderTextColor={COLORS.secondary}
        onChangeText={(text) => setPassword(text)}
        value={password}
      />
      <TouchableOpacity
        disabled={!canLogin}
        style={{
          ...styles.buttons,
          backgroundColor: canLogin ? COLORS.primary : COLORS.accent,
        }}
        onPress={login}
      >
        <Text
          style={{
            ...FONTS.p_regular,
            color: canLogin ? COLORS.accent : COLORS.secondary,
          }}
        >
          LOGIN
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        disabled={!canRegister}
        style={{
          ...styles.buttons,
          backgroundColor: canRegister ? COLORS.primary : COLORS.accent,
        }}
        onPress={register}
      >
        <Text
          style={{
            ...FONTS.p_regular,
            color: canRegister ? COLORS.accent : COLORS.secondary,
          }}
        >
          REGISTER
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = new StyleSheet.create({
  page: {
    height: "100%",
    backgroundColor: COLORS.secondary,
    paddingTop: "25%",
    alignItems: "center",
    paddingHorizontal: SIZES.padding,
  },
  textInput: {
    ...FONTS.h2_bold,
    backgroundColor: COLORS.accent,
    width: "100%",
    // color: COLORS.secondary,
    marginVertical: 10,
    paddingHorizontal: "10%",
    paddingVertical: SIZES.padding,
    borderRadius: SIZES.textBoxRadius,
    borderWidth: 8,
    // borderColor: COLORS.primary,
  },
  buttons: {
    ...SHADOW,
    width: "50%",
    marginHorizontal: "25%",
    marginTop: "7%",
    padding: SIZES.padding,
    alignItems: "center",
    borderRadius: SIZES.textBoxRadius,
  },
  activity: {
    position: "absolute",
    top: 0,
    right: 0,
    height: Dimensions.get("window").height / 1.5,
    width: Dimensions.get("window").width,
    alignItems: "center",
    justifyContent: "center",
    elevation: 5,
  },
});
