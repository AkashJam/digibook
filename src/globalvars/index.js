import API from "./api";
import { UserContext, UserProvider, toastr } from "./store";

export { toastr, UserContext, UserProvider, API };
