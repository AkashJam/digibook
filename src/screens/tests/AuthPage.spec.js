import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import AuthPage from "../AuthPage";
import { UserProvider } from "../../globalvars";

// to mock login function fetch
require("jest-fetch-mock").enableMocks();
fetchMock.dontMock();

jest.mock("react-native-root-toast", () => {
  const { View } = require("react-native");
  return {
    Toast: View,
  };
});

jest.mock("expo-constants", () => {
  const Constants = { isDevice: false };
  return Constants;
});

const mockedNavigate = jest.fn();

const navigation = {
  navigate: mockedNavigate,
};

describe("AuthPage", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <UserProvider>
          <AuthPage />
        </UserProvider>
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it("username input field exists", () => {
    const { queryByPlaceholderText } = render(
      <UserProvider>
        <AuthPage />
      </UserProvider>
    );
    expect(queryByPlaceholderText("Username")).not.toBeNull();
  });

  it("password input field exists", () => {
    const { queryByPlaceholderText } = render(
      <UserProvider>
        <AuthPage />
      </UserProvider>
    );
    expect(queryByPlaceholderText("Password")).not.toBeNull();
  });

  it("login button initially disabled", () => {
    const { getByText } = render(
      <UserProvider>
        <AuthPage navigation={navigation} />
      </UserProvider>
    );
    fireEvent.press(getByText("LOGIN"));
    expect(mockedNavigate).not.toHaveBeenCalled();
  });

  it("register button initially disabled", () => {
    const { getByText } = render(
      <UserProvider>
        <AuthPage navigation={navigation} />
      </UserProvider>
    );
    fireEvent.press(getByText("REGISTER"));
    expect(mockedNavigate).not.toHaveBeenCalled();
  });

  it("test against snapshot", () => {
    const tree = renderer
      .create(
        <UserProvider>
          <AuthPage />
        </UserProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
