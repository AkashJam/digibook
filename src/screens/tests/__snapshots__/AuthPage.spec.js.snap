// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage test against snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-i7nevx r-height-1pi2tsx r-paddingHorizontal-1x7blby r-paddingTop-1gingfq"
>
  <div
    className="css-text-901oao r-color-18or775"
    dir="auto"
    style={
      Object {
        "fontFamily": "Caviar_Dreams_Bold",
        "fontSize": "24px",
        "paddingBottom": "4%",
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "paddingTop": "4%",
      }
    }
  >
    Digital Notebook
  </div>
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-11aywtz r-placeholderTextColor-1o1bpf7"
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onSelect={[Function]}
    placeholder="Username"
    readOnly={false}
    spellCheck={true}
    style={
      Object {
        "borderBottomColor": "rgba(0,166,251,1.00)",
        "borderLeftColor": "rgba(0,166,251,1.00)",
        "borderRightColor": "rgba(0,166,251,1.00)",
        "borderTopColor": "rgba(0,166,251,1.00)",
      }
    }
    type="text"
    value=""
  />
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-11aywtz r-placeholderTextColor-1o1bpf7"
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onSelect={[Function]}
    placeholder="Password"
    readOnly={false}
    spellCheck={true}
    style={
      Object {
        "borderBottomColor": "rgba(0,166,251,1.00)",
        "borderLeftColor": "rgba(0,166,251,1.00)",
        "borderRightColor": "rgba(0,166,251,1.00)",
        "borderTopColor": "rgba(0,166,251,1.00)",
      }
    }
    type="password"
    value=""
  />
  <div
    aria-disabled={true}
    className="css-view-1dbjc4n r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "WebkitTransitionDuration": "0s",
        "backgroundColor": "rgba(246,240,237,1.00)",
        "transitionDuration": "0s",
      }
    }
    tabIndex="-1"
  >
    <div
      className="css-text-901oao r-fontFamily-1cdzzfu r-fontSize-1i10wst"
      dir="auto"
      style={
        Object {
          "color": "rgba(54,54,53,1.00)",
        }
      }
    >
      LOGIN
    </div>
  </div>
  <div
    aria-disabled={true}
    className="css-view-1dbjc4n r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "WebkitTransitionDuration": "0s",
        "backgroundColor": "rgba(246,240,237,1.00)",
        "transitionDuration": "0s",
      }
    }
    tabIndex="-1"
  >
    <div
      className="css-text-901oao r-fontFamily-1cdzzfu r-fontSize-1i10wst"
      dir="auto"
      style={
        Object {
          "color": "rgba(54,54,53,1.00)",
        }
      }
    >
      REGISTER
    </div>
  </div>
  <div
    className="css-text-901oao r-color-18or775 r-fontFamily-1cdzzfu r-fontSize-1i10wst r-paddingHorizontal-cf760m r-paddingVertical-1crsq4q r-textAlign-q4m81j"
    dir="auto"
  >
    First time? Get started with just a username and password
  </div>
</div>
`;
